ocaml/typing/primitive.mli

```diff
@@ -51,12 +51,12 @@ type description = private
     prim_effects: effects;
     prim_coeffects: coeffects;
     prim_native_name: string;  (* Name of C function for the nat. code gen. *)
-    prim_native_repr_args: (mode * native_repr) list;
-    prim_native_repr_res: mode * native_repr }
+    prim_native_repr_args: (mode * Layouts.sort * native_repr) list;
```

Comment:
I agree it would be reasonable to give this tuple a name (or, even, make it a record!).  But I think I'll wait for @mshinwell or @Ekdohibs to weigh in on the previous question about whether we should have a 3-tuple at all.

Reply:
Well, there are two things to note about this:
- the sort is actually needed in bytecode mode (not immediately, but if we ever get primitives that have unboxed records/variants as arguments or results, it will definitely be needed), and I'd like it not to be in a field with `native` in its name
- I'd like this to be removed in the medium term, leaving only the sort, to simplify the code, expanding the primitive using box/unbox in the lambda translation when compiling in native mode
Given the above points, I think the best thing to do is either add a new field for the sorts or reuse prim_native_repr_args for now, but making the tuple into a new type declaration or a record is probably not necessary if we plan to remove this code not too long from now.

Reply:
This is just a style point. Resolving.